### PR TITLE
Fix palette images for float data

### LINKED ISSUE #2371: cloud_top_height colormap is broken
**Describe the bug**

The cloud_top_height composite is broken.  The resulting image looks wrong.  The color table written to GeoTIFF is wrong.

**To Reproduce**

```python
from satpy import Scene
from sattools.io import plotdir
sc = Scene(
    filenames=["/media/nas/x21308/scratch/NWCSAF/NWCSAF-NinjoTIFF-3/S_NWC_CTTH_MSG4_MSG-N-VISIR_20230131T083000Z.nc"],
    reader=["nwcsaf-geo"])
sc.load(["cloud_top_height"])
sc.save_datasets(filename=str(plotdir() / "test-palette.tif"), keep_palette=True)
```

**Expected behavior**

I expect an image like this:

oflxs489-METEOSATNOWCSAF_EUROPA_GESAMT_CTH_nqeuro3km_202301311245

which has a colourtable like this:

<details><summary>gdalinfo expected</summary>

```
Driver: GTiff/GeoTIFF
Files: oflxs489-METEOSATNOWCSAF_EUROPA_GESAMT_CTH_nqeuro3km_202301311245.tif
Size is 4000, 2000
Origin = (-53.989609999999992,76.988059999999990)
Pixel Size = (0.026994805000000,-0.026988060000000)
Metadata:
  TIFFTAG_DATETIME=2023:01:31 13:01:14
  TIFFTAG_SOFTWARE=tifffile/pytroll
Image Structure Metadata:
  COMPRESSION=DEFLATE
  INTERLEAVE=BAND
Corner Coordinates:
Upper Left  ( -53.9896100,  76.9880600) 
Lower Left  ( -53.9896100,  23.0119400) 
Upper Right (  53.9896100,  76.9880600) 
Lower Right (  53.9896100,  23.0119400) 
Center      (   0.0000000,  50.0000000) 
Band 1 Block=4000x2000 Type=Byte, ColorInterp=Palette
  Color Table (RGB with 256 entries)
    0: 0,0,0,255
    1: 254,0,127,255
    2: 254,0,215,255
    3: 125,0,42,255
    4: 152,19,46,255
    5: 177,50,0,255
    6: 254,75,0,255
    7: 254,101,0,255
    8: 254,163,0,255
    9: 254,215,0,255
   10: 254,254,0,255
   11: 215,254,0,255
   12: 177,254,0,255
   13: 152,254,0,255
   14: 0,254,0,255
   15: 0,139,47,255
   16: 0,177,254,255
   17: 0,215,254,255
   18: 0,254,254,255
   19: 237,213,209,255
   20: 238,238,222,255
   21: 254,254,254,255
   22: 0,0,0,255
   23: 0,0,0,255
   24: 0,0,0,255
   25: 0,0,0,255
   26: 0,0,0,255
   27: 0,0,0,255
   28: 0,0,0,255
   29: 0,0,0,255
   30: 0,0,0,255
   31: 0,0,0,255
   32: 0,0,0,255
   33: 0,0,0,255
   34: 0,0,0,255
   35: 0,0,0,255
   36: 0,0,0,255
   37: 0,0,0,255
   38: 0,0,0,255
   39: 0,0,0,255
   40: 0,0,0,255
   41: 0,0,0,255
   42: 0,0,0,255
   43: 0,0,0,255
   44: 0,0,0,255
   45: 0,0,0,255
   46: 0,0,0,255
   47: 0,0,0,255
   48: 0,0,0,255
   49: 0,0,0,255
   50: 0,0,0,255
   51: 0,0,0,255
   52: 0,0,0,255
   53: 0,0,0,255
   54: 0,0,0,255
   55: 0,0,0,255
   56: 0,0,0,255
   57: 0,0,0,255
   58: 0,0,0,255
   59: 0,0,0,255
   60: 0,0,0,255
   61: 0,0,0,255
   62: 0,0,0,255
   63: 0,0,0,255
   64: 0,0,0,255
   65: 0,0,0,255
   66: 0,0,0,255
   67: 0,0,0,255
   68: 0,0,0,255
   69: 0,0,0,255
   70: 0,0,0,255
   71: 0,0,0,255
   72: 0,0,0,255
   73: 0,0,0,255
   74: 0,0,0,255
   75: 0,0,0,255
   76: 0,0,0,255
   77: 0,0,0,255
   78: 0,0,0,255
   79: 0,0,0,255
   80: 0,0,0,255
   81: 0,0,0,255
   82: 0,0,0,255
   83: 0,0,0,255
   84: 0,0,0,255
   85: 0,0,0,255
   86: 0,0,0,255
   87: 0,0,0,255
   88: 0,0,0,255
   89: 0,0,0,255
   90: 0,0,0,255
   91: 0,0,0,255
   92: 0,0,0,255
   93: 0,0,0,255
   94: 0,0,0,255
   95: 0,0,0,255
   96: 0,0,0,255
   97: 0,0,0,255
   98: 0,0,0,255
   99: 0,0,0,255
  100: 0,0,0,255
  101: 0,0,0,255
  102: 0,0,0,255
  103: 0,0,0,255
  104: 0,0,0,255
  105: 0,0,0,255
  106: 0,0,0,255
  107: 0,0,0,255
  108: 0,0,0,255
  109: 0,0,0,255
  110: 0,0,0,255
  111: 0,0,0,255
  112: 0,0,0,255
  113: 0,0,0,255
  114: 0,0,0,255
  115: 0,0,0,255
  116: 0,0,0,255
  117: 0,0,0,255
  118: 0,0,0,255
  119: 0,0,0,255
  120: 0,0,0,255
  121: 0,0,0,255
  122: 0,0,0,255
  123: 0,0,0,255
  124: 0,0,0,255
  125: 0,0,0,255
  126: 0,0,0,255
  127: 0,0,0,255
  128: 0,0,0,255
  129: 0,0,0,255
  130: 0,0,0,255
  131: 0,0,0,255
  132: 0,0,0,255
  133: 0,0,0,255
  134: 0,0,0,255
  135: 0,0,0,255
  136: 0,0,0,255
  137: 0,0,0,255
  138: 0,0,0,255
  139: 0,0,0,255
  140: 0,0,0,255
  141: 0,0,0,255
  142: 0,0,0,255
  143: 0,0,0,255
  144: 0,0,0,255
  145: 0,0,0,255
  146: 0,0,0,255
  147: 0,0,0,255
  148: 0,0,0,255
  149: 0,0,0,255
  150: 0,0,0,255
  151: 0,0,0,255
  152: 0,0,0,255
  153: 0,0,0,255
  154: 0,0,0,255
  155: 0,0,0,255
  156: 0,0,0,255
  157: 0,0,0,255
  158: 0,0,0,255
  159: 0,0,0,255
  160: 0,0,0,255
  161: 0,0,0,255
  162: 0,0,0,255
  163: 0,0,0,255
  164: 0,0,0,255
  165: 0,0,0,255
  166: 0,0,0,255
  167: 0,0,0,255
  168: 0,0,0,255
  169: 0,0,0,255
  170: 0,0,0,255
  171: 0,0,0,255
  172: 0,0,0,255
  173: 0,0,0,255
  174: 0,0,0,255
  175: 0,0,0,255
  176: 0,0,0,255
  177: 0,0,0,255
  178: 0,0,0,255
  179: 0,0,0,255
  180: 0,0,0,255
  181: 0,0,0,255
  182: 0,0,0,255
  183: 0,0,0,255
  184: 0,0,0,255
  185: 0,0,0,255
  186: 0,0,0,255
  187: 0,0,0,255
  188: 0,0,0,255
  189: 0,0,0,255
  190: 0,0,0,255
  191: 0,0,0,255
  192: 0,0,0,255
  193: 0,0,0,255
  194: 0,0,0,255
  195: 0,0,0,255
  196: 0,0,0,255
  197: 0,0,0,255
  198: 0,0,0,255
  199: 0,0,0,255
  200: 0,0,0,255
  201: 0,0,0,255
  202: 0,0,0,255
  203: 0,0,0,255
  204: 0,0,0,255
  205: 0,0,0,255
  206: 0,0,0,255
  207: 0,0,0,255
  208: 0,0,0,255
  209: 0,0,0,255
  210: 0,0,0,255
  211: 0,0,0,255
  212: 0,0,0,255
  213: 0,0,0,255
  214: 0,0,0,255
  215: 0,0,0,255
  216: 0,0,0,255
  217: 0,0,0,255
  218: 0,0,0,255
  219: 0,0,0,255
  220: 0,0,0,255
  221: 0,0,0,255
  222: 0,0,0,255
  223: 0,0,0,255
  224: 0,0,0,255
  225: 0,0,0,255
  226: 0,0,0,255
  227: 0,0,0,255
  228: 0,0,0,255
  229: 0,0,0,255
  230: 0,0,0,255
  231: 0,0,0,255
  232: 0,0,0,255
  233: 0,0,0,255
  234: 0,0,0,255
  235: 0,0,0,255
  236: 0,0,0,255
  237: 0,0,0,255
  238: 0,0,0,255
  239: 0,0,0,255
  240: 0,0,0,255
  241: 0,0,0,255
  242: 0,0,0,255
  243: 0,0,0,255
  244: 0,0,0,255
  245: 0,0,0,255
  246: 0,0,0,255
  247: 0,0,0,255
  248: 0,0,0,255
  249: 0,0,0,255
  250: 0,0,0,255
  251: 0,0,0,255
  252: 0,0,0,255
  253: 0,0,0,255
  254: 0,0,0,255
  255: 0,0,0,255
```
</details>

**Actual results**

Instead, Satpy produces an image like this:

oflxs494-METEOSATNOWCSAF_EUROPA_GESAMT_CTH_nqeuro3km_202301311245

with an incorrect colortable:

<details><summary>Actual gdalinfo output</summary>

```
Driver: GTiff/GeoTIFF
Files: oflxs494-METEOSATNOWCSAF_EUROPA_GESAMT_CTH_nqeuro3km_202301311245.tif
Size is 4000, 2000
Coordinate System is:
PROJCRS["WGS 84 / World Equidistant Cylindrical",
    BASEGEOGCRS["WGS 84",
        ENSEMBLE["World Geodetic System 1984 ensemble",
            MEMBER["World Geodetic System 1984 (Transit)"],
            MEMBER["World Geodetic System 1984 (G730)"],
            MEMBER["World Geodetic System 1984 (G873)"],
            MEMBER["World Geodetic System 1984 (G1150)"],
            MEMBER["World Geodetic System 1984 (G1674)"],
            MEMBER["World Geodetic System 1984 (G1762)"],
            MEMBER["World Geodetic System 1984 (G2139)"],
            ELLIPSOID["WGS 84",6378137,298.257223563,
                LENGTHUNIT["metre",1]],
            ENSEMBLEACCURACY[2.0]],
        PRIMEM["Greenwich",0,
            ANGLEUNIT["degree",0.0174532925199433]],
        ID["EPSG",4326]],
    CONVERSION["World Equidistant Cylindrical",
        METHOD["Equidistant Cylindrical",
            ID["EPSG",1028]],
        PARAMETER["Latitude of 1st standard parallel",0,
            ANGLEUNIT["degree",0.0174532925199433],
            ID["EPSG",8823]],
        PARAMETER["Longitude of natural origin",0,
            ANGLEUNIT["degree",0.0174532925199433],
            ID["EPSG",8802]],
        PARAMETER["False easting",0,
            LENGTHUNIT["metre",1],
            ID["EPSG",8806]],
        PARAMETER["False northing",0,
            LENGTHUNIT["metre",1],
            ID["EPSG",8807]]],
    CS[Cartesian,2],
        AXIS["easting (X)",east,
            ORDER[1],
            LENGTHUNIT["metre",1]],
        AXIS["northing (Y)",north,
            ORDER[2],
            LENGTHUNIT["metre",1]],
    USAGE[
        SCOPE["Graticule coordinates expressed in simple Cartesian form."],
        AREA["World."],
        BBOX[-90,-180,90,180]],
    ID["EPSG",4087]]
Data axis to CRS axis mapping: 1,2
Origin = (-6010095.893327429890633,8570271.636361993849277)
Pixel Size = (3005.047946663714811,-3004.297096698314817)
Metadata:
  AREA_OR_POINT=Area
  ninjo_AxisIntercept=-0.0
  ninjo_CentralMeridian=0.0
  ninjo_ChannelID=Cloud top height
  ninjo_ColorDepth=8
  ninjo_CreationDateID=1675170136
  ninjo_DataSource=EUMETCAST
  ninjo_DataType=GPRN
  ninjo_DateID=1675169657
  ninjo_EarthRadiusLarge=6378137.0
  ninjo_EarthRadiusSmall=6356752.314245179
  ninjo_FileName=/data/pytroll/TMP/staging/METEOSATNOWCSAF_EUROPA_GESAMT_CTH_nqeuro3km_202301311245-geotiff.tif
  ninjo_Gradient=255.0
  ninjo_HeaderVersion=2
  ninjo_Magic=NINJO
  ninjo_MaxGrayValue=255
  ninjo_MinGrayValue=50
  ninjo_PhysicUnit=m
  ninjo_PhysicValue=Cloud top height
  ninjo_Projection=PLAT
  ninjo_ReferenceLatitude1=0.0
  ninjo_SatelliteNameID=NWCSAF Europe/Atlantic
  ninjo_TransparentPixel=0
  ninjo_XMaximum=4000
  ninjo_XMinimum=1
  ninjo_YMaximum=2000
  ninjo_YMinimum=1
  TIFFTAG_DATETIME=2023:01:31 12:54:17
Image Structure Metadata:
  COMPRESSION=LZW
  INTERLEAVE=BAND
  PREDICTOR=2
Corner Coordinates:
Upper Left  (-6010095.893, 8570271.636) ( 53d59'22.60"W, 76d59'17.02"N)
Lower Left  (-6010095.893, 2561677.443) ( 53d59'22.60"W, 23d 0'42.98"N)
Upper Right ( 6010095.893, 8570271.636) ( 53d59'22.60"E, 76d59'17.02"N)
Lower Right ( 6010095.893, 2561677.443) ( 53d59'22.60"E, 23d 0'42.98"N)
Center      (       0.000, 5565974.540) (  0d 0' 0.01"E, 50d 0' 0.00"N)
Band 1 Block=512x512 Type=Byte, ColorInterp=Palette
  NoData Value=0
  Color Table (RGB with 256 entries)
    0: 255,0,128,0
    1: 255,0,128,255
    2: 255,0,128,255
    3: 255,0,128,255
    4: 255,0,128,255
    5: 255,0,128,255
    6: 255,0,128,255
    7: 255,0,128,255
    8: 255,0,128,255
    9: 255,0,128,255
   10: 255,0,128,255
   11: 255,0,128,255
   12: 255,0,128,255
   13: 255,0,128,255
   14: 255,0,128,255
   15: 255,0,216,255
   16: 255,0,216,255
   17: 255,0,216,255
   18: 255,0,216,255
   19: 126,0,43,255
   20: 126,0,43,255
   21: 126,0,43,255
   22: 126,0,43,255
   23: 153,20,47,255
   24: 153,20,47,255
   25: 153,20,47,255
   26: 153,20,47,255
   27: 178,51,0,255
   28: 178,51,0,255
   29: 178,51,0,255
   30: 178,51,0,255
   31: 255,76,0,255
   32: 255,76,0,255
   33: 255,76,0,255
   34: 255,76,0,255
   35: 255,76,0,255
   36: 255,102,0,255
   37: 255,102,0,255
   38: 255,102,0,255
   39: 255,102,0,255
   40: 255,164,0,255
   41: 255,164,0,255
   42: 255,164,0,255
   43: 255,164,0,255
   44: 255,164,0,255
   45: 255,164,0,255
   46: 255,216,0,255
   47: 255,216,0,255
   48: 255,216,0,255
   49: 255,216,0,255
   50: 255,216,0,255
   51: 255,255,0,255
   52: 255,255,0,255
   53: 255,255,0,255
   54: 255,255,0,255
   55: 255,255,0,255
   56: 216,255,0,255
   57: 216,255,0,255
   58: 216,255,0,255
   59: 216,255,0,255
   60: 216,255,0,255
   61: 216,255,0,255
   62: 216,255,0,255
   63: 178,255,0,255
   64: 178,255,0,255
   65: 178,255,0,255
   66: 178,255,0,255
   67: 178,255,0,255
   68: 178,255,0,255
   69: 153,255,0,255
   70: 153,255,0,255
   71: 153,255,0,255
   72: 153,255,0,255
   73: 153,255,0,255
   74: 153,255,0,255
   75: 153,255,0,255
   76: 0,255,0,255
   77: 0,255,0,255
   78: 0,255,0,255
   79: 0,255,0,255
   80: 0,255,0,255
   81: 0,255,0,255
   82: 0,255,0,255
   83: 0,140,48,255
   84: 0,140,48,255
   85: 0,140,48,255
   86: 0,140,48,255
   87: 0,140,48,255
   88: 0,140,48,255
   89: 0,140,48,255
   90: 0,140,48,255
   91: 0,140,48,255
   92: 0,178,255,255
   93: 0,178,255,255
   94: 0,178,255,255
   95: 0,178,255,255
   96: 0,178,255,255
   97: 0,178,255,255
   98: 0,178,255,255
   99: 0,178,255,255
  100: 0,178,255,255
  101: 0,216,255,255
  102: 0,216,255,255
  103: 0,216,255,255
  104: 0,216,255,255
  105: 0,216,255,255
  106: 0,216,255,255
  107: 0,216,255,255
  108: 0,216,255,255
  109: 0,216,255,255
  110: 0,216,255,255
  111: 0,216,255,255
  112: 0,255,255,255
  113: 0,255,255,255
  114: 0,255,255,255
  115: 0,255,255,255
  116: 0,255,255,255
  117: 0,255,255,255
  118: 0,255,255,255
  119: 0,255,255,255
  120: 0,255,255,255
  121: 0,255,255,255
  122: 0,255,255,255
  123: 0,255,255,255
  124: 238,214,210,255
  125: 238,214,210,255
  126: 238,214,210,255
  127: 238,214,210,255
  128: 238,214,210,255
  129: 238,214,210,255
  130: 238,214,210,255
  131: 238,214,210,255
  132: 238,214,210,255
  133: 238,214,210,255
  134: 238,214,210,255
  135: 238,214,210,255
  136: 238,214,210,255
  137: 238,214,210,255
  138: 238,214,210,255
  139: 238,214,210,255
  140: 238,214,210,255
  141: 238,214,210,255
  142: 238,214,210,255
  143: 239,239,223,255
  144: 239,239,223,255
  145: 239,239,223,255
  146: 239,239,223,255
  147: 239,239,223,255
  148: 239,239,223,255
  149: 239,239,223,255
  150: 239,239,223,255
  151: 239,239,223,255
  152: 239,239,223,255
  153: 239,239,223,255
  154: 239,239,223,255
  155: 239,239,223,255
  156: 239,239,223,255
  157: 239,239,223,255
  158: 239,239,223,255
  159: 239,239,223,255
  160: 239,239,223,255
  161: 239,239,223,255
  162: 255,255,255,255
  163: 255,255,255,255
  164: 255,255,255,255
  165: 255,255,255,255
  166: 255,255,255,255
  167: 255,255,255,255
  168: 255,255,255,255
  169: 255,255,255,255
  170: 255,255,255,255
  171: 255,255,255,255
  172: 255,255,255,255
  173: 255,255,255,255
  174: 255,255,255,255
  175: 255,255,255,255
  176: 255,255,255,255
  177: 255,255,255,255
  178: 255,255,255,255
  179: 255,255,255,255
  180: 255,255,255,255
  181: 255,255,255,255
  182: 255,255,255,255
  183: 255,255,255,255
  184: 255,255,255,255
  185: 255,255,255,255
  186: 255,255,255,255
  187: 255,255,255,255
  188: 255,255,255,255
  189: 255,255,255,255
  190: 255,255,255,255
  191: 255,255,255,255
  192: 255,255,255,255
  193: 255,255,255,255
  194: 255,255,255,255
  195: 255,255,255,255
  196: 255,255,255,255
  197: 255,255,255,255
  198: 255,255,255,255
  199: 255,255,255,255
  200: 255,255,255,255
  201: 255,255,255,255
  202: 255,255,255,255
  203: 255,255,255,255
  204: 255,255,255,255
  205: 255,255,255,255
  206: 255,255,255,255
  207: 255,255,255,255
  208: 255,255,255,255
  209: 0,0,0,255
  210: 0,0,0,255
  211: 0,0,0,255
  212: 0,0,0,255
  213: 0,0,0,255
  214: 0,0,0,255
  215: 0,0,0,255
  216: 0,0,0,255
  217: 0,0,0,255
  218: 0,0,0,255
  219: 0,0,0,255
  220: 0,0,0,255
  221: 0,0,0,255
  222: 0,0,0,255
  223: 0,0,0,255
  224: 0,0,0,255
  225: 0,0,0,255
  226: 0,0,0,255
  227: 0,0,0,255
  228: 0,0,0,255
  229: 0,0,0,255
  230: 0,0,0,255
  231: 0,0,0,255
  232: 0,0,0,255
  233: 0,0,0,255
  234: 0,0,0,255
  235: 0,0,0,255
  236: 0,0,0,255
  237: 0,0,0,255
  238: 0,0,0,255
  239: 0,0,0,255
  240: 0,0,0,255
  241: 0,0,0,255
  242: 0,0,0,255
  243: 0,0,0,255
  244: 0,0,0,255
  245: 0,0,0,255
  246: 0,0,0,255
  247: 0,0,0,255
  248: 0,0,0,255
  249: 0,0,0,255
  250: 0,0,0,255
  251: 0,0,0,255
  252: 0,0,0,255
  253: 0,0,0,255
  254: 0,0,0,255
  255: 0,0,0,255
```
</details>

**Environment Info:**

 - Satpy Version: main (v0.39.0-272-gee5a1abc6)
 - trollimage version: main (v1.20.0-2-g0c7bbad)

**Additional context**

There appears to be some undesirable stretching going on.

I introduced this bug in #2301.

I've also tested `cloudtype`, `convective_rain_rate`, and `precipitation_probability`.  Those all look as expected.

### PR DESCRIPTION
**Pull request by gerritholl**
Fix broken palette images for those datasets where the valid range is not 0 to 255.  When defining a composite with a palettize enhancement where the palette is read from the source file, one **should not** set `min_value` and `max_value` in the enhancement configuration!  If those are set and differ from the values defined in `valid_range` in the dataset to which the palette is being applied, the images will be wrong.  If `min_value` and `max_value` are set at all, they have to be equal to the values defined in `valid_range`, or the resulting colours will be wrong.  The correct and safe thing to do is to not set `min_value` and `max_value` at all.

 - [x] Closes #2371 
 - [x] Tests added 
 - [x] Fully documented

### CODE REVIEW

**Review by djhoese — APPROVED** [review]
After a quick glance this seems good to me. I'm not sure I should have the final word so I'd appreciate another review. If no one has time for a review then we can just merge this.

File: satpy/etc/enhancements/generic.yaml
Comment (mraspaud) [inline satpy/etc/enhancements/generic.yaml:363]: If we have a section about palettes in the documentation, I think it would be nice to add these lines to it

**Review by mraspaud — APPROVED** [review]
LGTM, thanks debugging this! I just have a tiny comment.

File: satpy/etc/enhancements/generic.yaml
Comment (mraspaud) [reply satpy/etc/enhancements/generic.yaml:363]: perfect thanks! Merging

**Pull request merged**
